trogon/widgets/parameter_controls.py

```diff
@@ -305,33 +303,15 @@ def list_to_tuples(
             return MultiValueParamData.process_cli_option(collected_values)
 
     def get_control_method(
-        self, argument_type: Any
+        self, argument_type: Any, choices: Sequence[str] | None
     ) -> Callable[[Any, Text, bool, OptionSchema | ArgumentSchema, str], Widget]:
-        text_click_types = {
-            click.STRING,
-            click.FLOAT,
-            click.INT,
-            click.UUID,
-        }
-        text_types = (
-            click.Path,
-            click.File,
-            click.IntRange,
-            click.FloatRange,
-            click.types.FuncParamType,
-        )
+        if choices:
+            return partial(self.make_choice_control, choices=choices)
 
-        is_text_type = argument_type in text_click_types or isinstance(
-            argument_type, text_types
-        )
-        if is_text_type:
-            return self.make_text_control
-        elif argument_type == click.BOOL:
+        if argument_type is bool:
             return self.make_checkbox_control
-        elif isinstance(argument_type, click.types.Choice):
-            return partial(self.make_choice_control, choices=argument_type.choices)
-        else:
-            return self.make_text_control
+
+        return self.make_text_control
```

Comment:
This is much shorter, and roughly equal, but removes respect for click-specific types like `IntRange`.